edu-services/sections-service/sections-impl/sakai/impl/src/java/org/sakaiproject/component/section/sakai/SectionManagerImpl.java

```diff
@@ -688,7 +688,24 @@ public EnrollmentRecord joinSection(String sectionUuid, int maxSize) throws Role
 		// must be the same
 		List membershipInCategory = authzGroupService.getAuthzUserGroupIds(categorySections, userUid);
 		
-		if (!membershipInCategory.isEmpty()) {
+		List realMembershipInCategory=new ArrayList();
+		// Check membership of groups
+		
+		for(String msic: (List <String>) membershipInCategory)
+		{
+			String realmGroup=msic;
+			Group group = findGroup(realmGroup);
+			Member member = group.getMember(userUid);
+			if(member == null) {
+					if (log.isDebugEnabled()) log.debug("getAuthzUserGroupIds said " + userUid + " is member of " + group.getId() + " but getMember disagrees");
```

Comment:
replace with slf4j style log message:
`log.debug("getAuthzUserGroupIds said {} is member of {} but getMember disagrees", userUid, group.getId());`